server/wal/wal.go

```diff
@@ -942,6 +941,10 @@ func (w *WAL) Save(st raftpb.HardState, ents []raftpb.Entry) error {
 }
 
 func (w *WAL) SaveSnapshot(e walpb.Snapshot) error {
+	if err := walpb.ValidateSnapshotForWrite(&e); err != nil {
```

Comment:
Let's say a, b, c

And c as a follower runs this code against a as a leader with old etcd version. Would the follower reject the snapshot from old-versioned leader?

Reply:
I think there would be no problem as raft is speaking using `raftpb.Snapshot`

That always contains 
https://github.com/etcd-io/etcd/blob/102c1984444c0bc971f5bd0be49423840f6d4351/raft/raftpb/raft.proto#L33
with: 
https://github.com/etcd-io/etcd/blob/102c1984444c0bc971f5bd0be49423840f6d4351/raft/raftpb/raft.proto#L26. 

This validation just makes sure we never forgot to copy the content from raftpb.Snapshot to walpb.Snapshot. 
That cannonically happens in server/etcdserver/storage.go (line 59)

Reply:
@ptabor +1. Yeah, this is for saving the snap locally.